### ISSUE TITLE
Add pillar parameter to set the major/minor version of SaltStack repository

### ISSUE THREAD

**Issue by kakwa** [saltstack-formulas/salt-formula#293]
Hello,

It would be nice if there was a way to set the version of saltstack, leveraging the already present versioning in the repository:

For example the saltstack debian repositories provide the following versioned URL:
```bash
#pin to Major version
deb http://repo.saltstack.com/apt/debian/8/amd64/2016.11 jessie main

#pin to Minor release
deb http://repo.saltstack.com/apt/debian/8/amd64/archive/2016.11.2 jessie main
```

Adding a pillar and using it in the [map.jinja file](https://github.com/saltstack-formulas/salt-formula/blob/master/salt/map.jinja#L42) could do the job. This pillar can default to ```latest``` to retain the current behavior.

Example implementation:
```yaml
salt:
   - release: '2016.11'
```

```diff
--- a/salt/map.jinja
+++ b/salt/map.jinja
@@ -39,7 +39,7 @@ that differ from whats in defaults.yaml
 {% set os_family_map = salt['grains.filter_by']({
     'Debian':  {
       'pkgrepo': 'deb http://repo.saltstack.com/apt/' +
-      salt['grains.get']('os')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/amd64/latest ' + salt['grains.get']('oscodename') + ' main',
+      salt['grains.get']('os')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/amd64/' + salt['pillar.get']('salt:release', 'latest') + ' ' + salt['grains.get']('oscodename') + ' main',
       'key_url': 'https://repo.saltstack.com/apt/' + salt['grains.get']('os')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/amd64/latest/SALTSTACK-GPG-KEY.pub',
       'libgit2': 'libgit2-22',
       'gitfs': {
@@ -138,7 +138,7 @@ that differ from whats in defaults.yaml
   }, merge=salt['grains.filter_by']({
     'Ubuntu':  {
       'pkgrepo': 'deb http://repo.saltstack.com/apt/' +
-      salt['grains.get']('os')|lower + '/' + osrelease + '/amd64/latest ' + salt['grains.get']('oscodename') + ' main',
+      salt['grains.get']('os')|lower + '/' + osrelease + '/amd64/' + salt['pillar.get']('salt:release', 'latest') + ' ' + salt['grains.get']('oscodename') + ' main',
       'key_url': 'https://repo.saltstack.com/apt/' + salt['grains.get']('os')|lower + '/' + osrelease + '/amd64/latest/SALTSTACK-GPG-KEY.pub',
       'pygit2': 'python-pygit2',
       'gitfs': {
@@ -153,7 +153,7 @@ that differ from whats in defaults.yaml
     },
     'Raspbian': {
       'pkgrepo': 'deb http://repo.saltstack.com/apt/' +
-      salt['grains.get']('os_family')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/armhf/latest ' + salt['grains.get']('oscodename') + ' main',
+      salt['grains.get']('os_family')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/armhf/' + salt['pillar.get']('salt:release', 'latest') + ' ' + salt['grains.get']('oscodename') + ' main',
       'key_url': 'https://repo.saltstack.com/apt/' + salt['grains.get']('os_family')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/armhf/latest/SALTSTACK-GPG-KEY.pub',
     },
     'SmartOS': {
```

This implementation seems to work fine on Debian, but I don't have the infrastructure to test it with other distributions/OSes, and I'm not sure it respects the proper guide lines. I'm only providing it as an example.

**Comment by daks** [comment]
I would really like this feature too.

**Comment by aboe76** [comment]
could be combined with : https://github.com/saltstack-formulas/salt-formula/pull/291

**Comment by daks** [comment]
@aboe76 seems good for me

**Comment by aboe76** [comment]
@kakwa could you make a PR?

**Comment by daks** [comment]
@aboe76 isn't https://github.com/saltstack-formulas/salt-formula/pull/291 already good?

**Comment by aboe76** [comment]
It's missing some version options for the other salt stuff

Op vr 26 mei 2017 09:38 schreef Éric Veiras Galisson <
notifications@github.com>:

**Comment by kakwa** [comment]
@aboe76 

here is the PR:
https://github.com/saltstack-formulas/salt-formula/pull/314

It needs some testing before merge, I don't have a testing environment for this.

**Comment by aboe76** [comment]
@kakwa I will test it tomorrow.

**Comment by partsalliance** [comment]
is this working for anyone? there is no change when i place release: '2016.3' inside my pillar file

**Comment by 0xf10e** [comment]
@partsalliance I'm sure I've pinned a subset of my minions to 2016.11 and I didn't pull any PR-branches (like #315) for this… Got to check my pillar.

**Comment by aboe76** [comment]
@partsalliance can you give us a redacted pillar file of the salt stuff to better understand your issue?
It should work with something like this:
```
salt:
  release: 2016.3
```

**Comment by aboe76** [comment]
close issue, fixed with #358
